cmake/CeleritasLibrary.cmake

```diff
@@ -83,6 +128,37 @@ define_property(TARGET PROPERTY CELERITAS_CUDA_OBJECT_LIBRARY
   FULL_DOCS "Name of the object (without nvlink step) library corresponding to this cuda library"
 )
 
+##############################################################################
+# Separate the OPTIONS out from the sources
+#
+macro(CUDA_GET_SOURCES_AND_OPTIONS _sources _cmake_options _options)
```

Comment:
macro/function names shouldn't be uppercase. If you want to use the same code and style as in VecGeom/Ben's code then include that file verbatim with attribution, and add wrappers for it.

Reply:
> VecGeom/Ben's 

Technically, it is code that was develop first (by me) within Celeritas (and the attribution is still [there](https://github.com/drbenmorgan/CudaRDCExercise/blob/main/CudaRdcUtils.cmake)) and I must have missed those kind of changes when Ben introduces his improvements.

The intent is for this code to be eventually submitted to `CMake` and thus should follow their convention.

> include that file verbatim with attribution

I do want to eventually get there, right now the only intended difference is the prefix (`celeritas_` vs `cuda_rdc`) and that 'only' a `sed` execution away so manageable.   For this round, especially since this pr would still be squashed, I would rather focus in functionality (on the ci failing is bad :( ) rather than spelling/casing/ifs-style which I could/will fix in a subsequent PR).